clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -180,19 +180,19 @@ CompletableFuture<ClientResponse> sendAutoCommit(final Map<TopicPartition, Offse
     private class OffsetCommitRequestState {
         private final Map<TopicPartition, OffsetAndMetadata> offsets;
         private final String groupId;
-        private final GroupState.Generation generation;
+        private final int memberEpoch;
```

Comment:
For my understanding, how do we plan to update the member epoch when stale member epoch is received? Are we going to recreate a new OffsetCommitRequestState?

Reply:
You are right. Stale member epcoh error isn't retriable, so the commit request would fail.  

Though - When we receive such error - It seems like we should also update membershipManager which is not done here.

Reply:
> Stale member epcoh error isn't retriable

A stale member epoch error is retriable... but it should be retried with the new epoch.

Reply:
We could do it either way - recreate this RequestState object or resend the same object with the updated epoch.